File: src_a_(C++)/Test4-3.c
//Проект Странник-Модула Для Windows 32, тестовая программа
//Группа тестов 4:ОПЕРАТОРЫ
//Тест номер    3:ОПЕРАТОР SWITCH

include Win32

char s[15];

int i,j;
void main() {
  i=4;
  j=0;
//1-я альтернатива
  switch (i) {
    case 4:j=1; break;
    case 5:j=2; break;
    default:j=3; break;
  }
  wvsprintf(s,"j=%li",&j);
  MessageBox(0,s,"j=1",0);
//2-я альтернатива
  switch (i) {
    case 2:j=1; break;
    case 3..5:j=2; break;
    default:j=3; break;
  }
  wvsprintf(s,"j=%li",&j);
  MessageBox(0,s,"j=2",0);
//ELSE часть
  switch (i) {
    case 2:j=1; break;
    case 3: case 5:j=2; break;
    default:j=3; break;
  }
  wvsprintf(s,"j=%li",&j);
  MessageBox(0,s,"j=3",0);
//отрицательная константа
  i=-1;
  switch (i) {
    case 2:j=1; break;
    case -1:j=2; break;
    case 3:j=3; break;
  }
  wvsprintf(s,"j=%li",&j);
  MessageBox(0,s,"j=2",0);
}

File: src_a_(Pascal)/Test4-3.pas
//Проект Странник-Модула Для Windows 32, тестовая программа
//Группа тестов 4:ОПЕРАТОРЫ
//Тест номер    3:ОПЕРАТОР CASE
program Test4_3;
uses Win32;

var s:string[15];

var i,j:integer;
begin
  i:=4;
  j:=0;
//1-я альтернатива
  case i of
    4:j:=1;
    5:j:=2;
  else j:=3
  end;
  wvsprintf(s,"j=%li",addr(j));
  MessageBox(0,s,"j=1",0);
//2-я альтернатива
  case i of
    2:j:=1;
    3..5:j:=2;
  else j:=3
  end;
  wvsprintf(s,"j=%li",addr(j));
  MessageBox(0,s,"j=2",0);
//ELSE часть
  case i of
    2:j:=1;
    3,5:j:=2;
  else j:=3
  end;
  wvsprintf(s,"j=%li",addr(j));
  MessageBox(0,s,"j=3",0);
//отрицательная константа
  i:=-1;
  case i of
    2:j:=1;
    -1:j:=2;
    3:j:=3;
  end;
  wvsprintf(s,"j=%li",addr(j));
  MessageBox(0,s,"j=2",0);
end.

